### DISCUSSION TITLE
Small help creating automatic install script for farmer

### DISCUSSION THREAD

**Discussion by forReason** [Chia-Network/chia-blockchain#8695] [Support]
Dear Folks,

I am expanding my farm and I'm looking to do a full automatic setup of my machines. 
So far so good, everything installs properly. With the chia client however, I encounter some user-input caviats with two configuration commands:

- chia keys add
- chia wallet show

chia keys add:  
This chia keys add requires user input such as   
```
chia keys add
# echo comes back: please enter 24 word mnemonic passphrase
hello this is an example mnemonic passphrase ... ... ... ... ...
```

chia wallet show:  
I use the chia wallet show command to actually start syncing the wallet. The main issue is, that plots do not farm on pool, when the wallet is not synchronized.  
right now the process is similar to this:  
```
chia wallet show
# echo comes back: No wallet backup file was found. To generate wallet, enter S
S
```

How can I automate these two commands in ubuntu .sh script?  
I know having my mnemonic keys clearly readable in a installation script file poses a risk. Right now to prevent this, I have a separate wallet for my farmer where up to a month of farmed chia resides. That risk is bearable for me. I will look into ways to secure that further but thats of no priority for my company right now.

**Comment by BrandtH22** [comment]
@forReason , since this ticket was opened a number of years ago without further comments it seems this issue has been resolved. 
We will be closing this ticket but if we have closed this ticket in error do not hesitate to reach out to us again with any followup questions or comments.

Note the second issue is likely tied to the first as there are no keys available for the wallet. To expedite the key creation you could just make a copy of the .chia_keys folder and have that added to the machines during the install script (as you noted this is not recommended due to potential security issues)

The best place to reach our support team is on Discord (https://discord.gg/chia) or by reopening this ticket.
